### DISCUSSION TITLE
[Bug]: value <select> incorrect when async loading single option

### DISCUSSION THREAD

**Discussion by sjbuysse** [react-hook-form/react-hook-form#6713] [Q&A]
### Version Number

7.15.3

### Codesandbox/Expo snack

https://stackblitz.com/edit/react-ts-bcqs6m

### Steps to reproduce

1. Go to https://stackblitz.com/edit/react-ts-bcqs6m
2. Wait for 1 second, so that the options of `product` select elements are loaded. Since there is only 1 option, the browser will automatically select this option. You will see option `1337` being selected
3. Press submit
4. Check the console, and notice that the `product` field has no value
(5. When inspecting the html element with chrome dev tools, you can see that the DOM element actually does hold the value, I added a screenshot to demonstrate this)
 <img width="1558" alt="Screenshot 2021-10-04 at 16 49 43" src="https://user-images.githubusercontent.com/12529291/135873577-7f022c24-e307-4419-8a55-8170a57332a0.png">

### Expected behaviour

The submitted data holds the value of the `select` element.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
Use `setValue` instead. it's a bit confusing example: i don't see anything on the screen as well.

<img width="1045" alt="Screen Shot 2021-10-05 at 10 25 02 am" src="https://user-images.githubusercontent.com/10513364/135937528-52ac2dd2-cc54-4398-aefb-612bf7c06eed.png">

if your select options are going to be updated with setState, then setValue will need to be invoked for update the selections.

**Comment by sjbuysse** [comment]
@bluebill1049 why is it confusing? The use-case is that you get your selection options async from a backend. 
When only 1 option is added, the DOM automatically selects it, this is default HTML behaviour. 
Yes, I could use `setValue`, but isn't it kind of a bug/issue that the state of the DOM `select` element is deviating from the one in our React app? 

I'm sorry the example isn't rendering for you, I wish it did, as it would make thing perhaps clearer. (for me it's running fine, by the way)
